data/transactions/logic/TEAL_opcodes.md

```diff
@@ -1560,3 +1560,45 @@ For boxes that exceed 4,096 bytes, consider `box_create`, `box_extract`, and `bo
 | 0 | BlkSeed | []byte |  |
 | 1 | BlkTimestamp | uint64 |  |
 
+
+## ec_add v
+
+- Opcode: 0xe0 {uint8 curve}
+- Stack: ..., A: []byte, B: []byte &rarr; ..., []byte
+- for (curve points A and B) return the curve point A + B
+- **Cost**:  BN254=0 BN254_G1=10 BN254_G2=10 BLS12_381=0 BLS12_381_G1=20 BLS12_381_G2=20
+- Availability: v9
+
+`EC` Curves:
+
+| Index | Name | Notes |
+| - | ------ | --------- |
+| 0 | BN254 | BN254 curve, sometimes called BN256 |
+| 1 | BN254_G1 | G1 of the BN254 curve |
+| 2 | BN254_G2 | G2 of the BN254 curve |
+| 3 | BLS12_381 | BLS 12-381 curve |
+| 4 | BLS12_381_G1 | G1 of the BLS 12-381 curve |
+| 5 | BLS12_381_G2 | G2 of the BLS 12-381 curve |
+
+
+A, B are curve points in G1 group. Each point consists of (X, Y) where X and Y are 256 bit integers, big-endian encoded. The encoded point is 64 bytes from concatenation of 32 byte X and 32 byte Y.
```

Comment:
```suggestion
A, B are curve points in affine representation: `X` concatenated to `Y`.

| Index | Name | Length of X or Y | Length of A or B |
| ----- | ------ | ---------------- | ----------------- |
| 1 | BN254_G1 | 32 bytes = 256 bits | 64 bytes = 512 bits |
| 2 | BN254_G2 | 64 bytes = 512 bits | 128 bytes = 1024 bits | 
| 4 | BLS12_381_G1 | 48 bytes = 384 bits | 96 bytes = 768 bits |
| 5 | BLS12_381_G2 | 96 bytes = 768 bits | 192 bytes = 1536 bits |


Checks:
* `ec_add` panics if `A` and `B` are not on the curve.
* `ec_add` does not verify that `A` and `B` are in the main prime-order subgroup.
```

Note for some reason https://eips.ethereum.org/EIPS/eip-2537 pads X/Y with 16 bytes for BLS12-381 G1 and with 32 bytes for G2. I don't think we need it.

Reply:
I think this is covered now.  It's annoying to make the table have different headings (they are generated). So the info is in the note field of the table.